pipelines/gitops-pull-request-rhtap/gitops-pull-request.yaml

```diff
@@ -66,10 +70,30 @@ spec:
       workspaces:
         - name: source
           workspace: workspace
+
+    - name: get-images-to-verify
+      runAfter:
+        - gather-deploy-images
+      params:
+        - name: INPUT
+          value: $(tasks.gather-deploy-images.results.IMAGES_TO_VERIFY)
+        - name: JQ_ARGS
+          value:
+            - --compact-output
+            # de-duplicate container images, drop 'env' attrs
```

Comment:
`gather-deploy-images` used to do the deduplication, but now we also need info about the environment because, if I remember correctly, we shouldn't upload SBOMs when an image gets promoted to the dev environment. Only stage or prod.

The question is, what should `gather-deploy-images` do when the same image gets promoted to multiple environment in a single PR
```json
[
  {"containerImage": "$image_1", "env": "development"},
  {"containerImage": "$image_1", "env": "stage"},
  {"containerImage": "$image_2", "env": "prod"}
]
```

I suppose it could de-duplicate them and prefer prod > stage > dev, so the result would be

```json
[
  {"containerImage": "$image_1", "env": "stage"},
  {"containerImage": "$image_2", "env": "prod"}
]
```

But that behavior seems weird, it should be up to the consumer how they want to process the result.

Reply:
But maybe I can get rid of `jq` by baking this logic into the `download-sboms` task. It will be less generic, but it's already strongly tied to RHTAP / Konflux because it looks for the SBOM_BLOB_URL result in the Tekton results in the attestation. So coupling it even more tightly isn't that big of a deal

Reply:
Alternatively, you can have an array in the `env` key.